src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/Miscellaneous/MarshalClassTests.cs

```diff
@@ -0,0 +1,188 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class MarshalClassTests
+    {
+#pragma warning disable 649
+        //definition of structure that will be used in testing of structs with Fixed BSTR Safearray fields
+        internal struct Variant
+        {
+            public ushort vt;
+            public ushort wReserved1;
+            public ushort wReserved2;
+            public ushort wReserved3;
+            public IntPtr bstrVal;
+            public IntPtr pRecInfo;
+        }
+#pragma warning restore 649
+
+        [StructLayout(LayoutKind.Auto)]
+        public struct SomeTestStruct_Auto
+        {
+            public int i;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct StructWithFxdLPSTRSAFld
+        {
+            [MarshalAs(UnmanagedType.ByValArray, ArraySubType = UnmanagedType.LPStr, SizeConst = 0)]
+            public String[] Arr;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct SomeTestStruct
+        {
+            public int i;
+            //[MarshalAs(UnmanagedType.BStr)]
+            public String s;
+        }
+
+        public enum TestEnum
+        {
+            red,
+            green,
+            blue
+        }
+
+        public struct TestStructWithEnumArray
+        {
+            [MarshalAs(UnmanagedType.ByValArray, SizeConst = 3)]
+            public TestEnum[] ArrayOfEnum;
+        }
+
+        public IntPtr ip;
+        public SomeTestStruct someTs = new SomeTestStruct();
+        public Exception ex = null;
+
+        [STAThread]
```

Comment:
Same question about STAThread; I'll stop commenting on it.

Reply:
Fixed, see in next commit